src/connection-manager/auto-dial.ts

```diff
@@ -103,6 +103,12 @@ export class AutoDial implements Startable {
 
     const connections = this.connectionManager.getConnectionsMap()
     const numConnections = connections.size
+    const dialQueue = new PeerSet(
+      // @ts-expect-error boolean filter removes falsy peer IDs
+      this.connectionManager.getDialQueue()
```

Comment:
type casting should work here and remove the need for ugly `//@ts-expect-error` s
```suggestion
 const dialQueue = new PeerSet(
      this.connectionManager.getDialQueue()
        .map(queue => queue.peerId)
        .filter(Boolean) as PeerId[]
    )
 ```

Reply:
The thing about the ugly `@ts-expect-error` is that should typescript ever be smart enough to work out what `.filter(Boolean)` does, it'll then become an error itself letting you know that the types are correctly inferred now whereas the `as` cast will smother the error forever.

Reply:
I see your point but wouldn't a `PeerId[]` type be expected here? I think the type error being returned by `.filter(Boolean)` is legitimate because all values _could_ be falsey, which in that case it would be more appropriate to have an empty array of `PeerId` passed to `new PeerSet` as opposed to `undefined`.

Reply:
This is a bug in typescript.  The issue reporting the bug is here: https://github.com/microsoft/TypeScript/issues/16655 - it was fixed by https://github.com/microsoft/TypeScript/pull/29955 and then reverted by https://github.com/microsoft/TypeScript/issues/16655#issuecomment-494907057 so remains unfixed.